v2/dht.go

```diff
@@ -302,21 +313,41 @@ func (d *DHT) setClientMode() {
 			}
 
 			if err := s.Reset(); err != nil {
-				d.log.With("err", err).Debug("failed closing stream")
+				d.debugErr(err, "failed closing stream")
 			}
 		}
 	}
 }
 
-// logErr is a helper method that uses the slogger of the DHT and writes a
-// warning log line with the given message alongside the error. If the error
+// warnErr is a helper method that uses the slogger of the DHT and writes a
+// warning log line with the given message alongside the error. args is a list of
+// key/value pairs or slog.Attrs that will be included with the log message. If the error
 // is nil, this method is a no-op.
-func (d *DHT) logErr(err error, msg string) {
+func (d *DHT) warnErr(err error, msg string, args ...any) {
 	if err == nil {
 		return
 	}
 
-	d.log.Warn(msg, "err", err.Error())
+	if len(args) == 0 {
+		d.log.Warn(msg, tele.LogAttrError(err))
+		return
+	}
+	d.log.With(args...).Warn(msg, tele.LogAttrError(err))
```

Comment:
not necessary, but now that we have helper methods for logging error messages we could use the more performant LogAttrs method here. Error logs shouldn't live in the hot path anyway so feel free to ignore.